spec/models/spree/payment_spec.rb

```diff
@@ -1,7 +1,691 @@
+            # Change it to something different
+            payment.response_code = 'abc'
+            payment.void_transaction!
+            expect(payment.response_code).to eq('12345')
+          end
+        end
+
+        context "if unsuccessful" do
+          it "should not void the payment" do
+            gateway.stub :void => failed_response
+            payment.should_not_receive(:void)
+            expect { payment.void_transaction! }.to raise_error(Spree::Core::GatewayError)
+          end
+        end
+
+        # Regression test for #2119
+        context "if payment is already voided" do
+          before do
+            payment.state = 'void'
+          end
+
+          it "should not void the payment" do
+            payment.payment_method.should_not_receive(:void)
+            payment.void_transaction!
+          end
+        end
+      end
+
+      context "#credit" do
+        before do
+          payment.state = 'completed'
+          payment.response_code = '123'
+        end
+
+        context "when outstanding_balance is less than payment amount" do
+          before do
+            payment.order.stub :outstanding_balance => 10
+            payment.stub :credit_allowed => 1000
+          end
+
+          it "should call credit on the gateway with the credit amount and response_code" do
+            gateway.should_receive(:credit).with(1000, card, '123', anything).and_return(success_response)
+            payment.credit!
+          end
+        end
+
+        context "when outstanding_balance is equal to payment amount" do
+          before do
+            payment.order.stub :outstanding_balance => payment.amount
+          end
+
+          it "should call credit on the gateway with the credit amount and response_code" do
+            gateway.should_receive(:credit).with(amount_in_cents, card, '123', anything).and_return(success_response)
+            payment.credit!
+          end
+        end
+
+        context "when outstanding_balance is greater than payment amount" do
+          before do
+            payment.order.stub :outstanding_balance => 101
+          end
+
+          it "should call credit on the gateway with the original payment amount and response_code" do
+            gateway.should_receive(:credit).with(amount_in_cents.to_f, card, '123', anything).and_return(success_response)
+            payment.credit!
+          end
+        end
+
+        it "should log the response" do
+          payment.credit!
+          expect(payment).to have_received(:record_response)
+        end
+
+        context "when gateway does not match the environment" do
+          it "should raise an exception" do
+            gateway.stub :environment => "foo"
+            expect { payment.credit! }.to raise_error(Spree::Core::GatewayError)
+          end
+        end
+
+        context "when response is successful" do
+          it "should create an offsetting payment" do
+            Spree::Payment.should_receive(:create!)
```

Comment:
this syntax can be modernized right? expect .... to receive.

Reply:
yeah, it should. Good catch.